### PR TITLE
fix: Add unidecode to integer input

### LINKED ISSUE #8627: pl-integer-input do full unidecode
pl-integer-input currently only handles the special case of U+2212, but not other possible unicode hyphens/minus (such as U+FF0D or U+FE63). This can cause issues if users are copying text from documents or other unicode enriched sources.

Add support for all forms of unicode hyphens and minuses, either by a lookup table or using [unidecode](https://pypi.org/project/Unidecode/).

### PR DESCRIPTION
**Pull request by eliotwrobson**
See title, adds helper function to resolve #8627

### CODE REVIEW

File: apps/prairielearn/python/prairielearn.py [hunk 1]
```diff
@@ -1812,3 +1813,8 @@ def index2key(i):
 
 def is_int_json_serializable(n: int) -> bool:
     return -((2**53) - 1) <= n <= 2**53 - 1
+
+
+def full_unidecode(input_str: str) -> str:
+    """Does unidecode of input"""
+    return unidecode(input_str.replace("\u2212", "-"))
```
Comment (jonatanschroeder) [inline apps/prairielearn/python/prairielearn.py:1820]: Should this be in prairielearn.py?

Comment (eliotwrobson) [reply apps/prairielearn/python/prairielearn.py:1820]: Do you mean the python helper sympy? I guess it would be a little awkward to import it from there but prevents some duplicate logic. I'll move it there on my next push.

Comment (jonatanschroeder) [reply apps/prairielearn/python/prairielearn.py:1820]: I'm just considering that it makes sense to have this normalization be consistent across elements.

Comment (eliotwrobson) [reply apps/prairielearn/python/prairielearn.py:1820]: Yes that makes sense, it's pretty easy to add this to the other elements once it's a helper function. I didn't do everything all at once just because I didn't have time to test every element.

Comment (eliotwrobson) [reply apps/prairielearn/python/prairielearn.py]: Switched to using uniform unidecode function here: https://github.com/PrairieLearn/PrairieLearn/pull/8634/commits/960769b0096267c004667233df31ee1718302445

Not sure if I'm in favor of switching every element over to this at once, one of those things that could cause weird minor issues I think. But easy enough to do one at a time now that this function is written.

Comment (jonatanschroeder) [reply apps/prairielearn/python/prairielearn.py]: I just realized that this was indeed in prairielearn.py to begin with, sorry for the confusion, my first read I thought this was in an element... In any case, I agree with switching elements gradually.

My comment was just in the sense of ensuring that any element that *does* have normalization does it consistently. I don't care where the function lives, as long as all elements that do it call a centralized function for it. This would also allow us to more easily expand to other characters (e.g., [`\u2215`](https://www.compart.com/en/unicode/U+2215) is also a convertible character that is not supported by text-unidecode; while [`\u03c0`](https://www.compart.com/en/unicode/U+03C0) could potentially be converted to `pi`) in the future and have that change affect all elements that do the conversion.

Comment (eliotwrobson) [reply apps/prairielearn/python/prairielearn.py]: I definitely agree with that reasoning. The way this is currently set up in this PR should allow for this consistent normalization.

File: apps/prairielearn/python/python_helper_sympy.py [hunk 2]
```diff
@@ -703,3 +701,8 @@ def get_items_list(items_string: None | str) -> list[str]:
         return []
 
     return list(map(str.strip, items_string.split(",")))
+
+
+def full_unidecode(input_str: str) -> str:
+    """Does unidecode of input and replaces the unicode minus with the normal one."""
+    return unidecode(input_str.replace("\u2212", "-"))
```
Comment (nwalters512) [inline apps/prairielearn/python/python_helper_sympy.py:708]: This does not feel like it belongs here - there is absolutely nothing sympy-specific about this function. It looks like you had it in `prairielearn.py` in a previous iteration; why the change?

Comment (eliotwrobson) [reply apps/prairielearn/python/python_helper_sympy.py:708]: If put in prairielearn.py, it can't be imported here because of a circular import.

Comment (nwalters512) [reply apps/prairielearn/python/python_helper_sympy.py:708]: Hmm, that's unfortunate. I'd rather use a workaround like doing `from prairielearn import full_unidecode` inside `evaluate_with_source`.

Comment (eliotwrobson) [reply apps/prairielearn/python/python_helper_sympy.py]: In https://github.com/PrairieLearn/PrairieLearn/pull/8634/commits/66310efceeeac915c8199a5682f1631348e60d15 I was able to avoid this by importing the modules (I think).

Comment (nwalters512) [reply apps/prairielearn/python/python_helper_sympy.py]: Looks like Black needs to reformat some things.

**Review by nwalters512 — APPROVED** [review]
